src/pages/iou/request/step/IOURequestStepMerchant.js

```diff
@@ -25,25 +33,48 @@ const propTypes = {
     /** Onyx Props */
     /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
     transaction: transactionPropTypes,
+
+    /** The draft transaction that holds data to be persisted on the current transaction */
+    splitDraftTransaction: transactionPropTypes,
+
+    /** The policy of the report */
+    policy: policyPropTypes.policy,
+
+    /** Collection of categories attached to a policy */
+    policyCategories: PropTypes.objectOf(categoryPropTypes),
+
+    /** Collection of tags attached to a policy */
+    policyTags: tagPropTypes,
 };
 
 const defaultProps = {
     transaction: {},
+    splitDraftTransaction: {},
+    policy: null,
+    policyTags: null,
+    policyCategories: null,
 };
 
 function IOURequestStepMerchant({
     route: {
-        params: {transactionID, backTo},
+        params: {transactionID, reportID, backTo, action, iouType},
     },
-    transaction: {merchant, participants},
+    transaction,
+    splitDraftTransaction,
+    policy,
+    policyTags,
+    policyCategories,
 }) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {inputCallbackRef} = useAutoFocusInput();
-    const isEmptyMerchant = merchant === '' || merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
-
-    const isMerchantRequired = _.some(participants, (participant) => Boolean(participant.isPolicyExpenseChat));
+    const isEditing = action === CONST.IOU.ACTION.EDIT;
 
+    // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
+    const isEditingSplitBill = iouType === CONST.IOU.TYPE.SPLIT && isEditing;
+    const {merchant} = ReportUtils.getTransactionDetails(isEditingSplitBill && !lodashIsEmpty(splitDraftTransaction) ? splitDraftTransaction : transaction);
+    const isEmptyMerchant = merchant === '' || merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
+    const isMerchantRequired = _.some(transaction.participants, (participant) => Boolean(participant.isPolicyExpenseChat));
```

Comment:
@cead22 It is old logic
https://github.com/Expensify/App/blob/1f1867a0a969c91e2169603b33ea272067428504/src/pages/iou/request/step/IOURequestStepMerchant.js#L45

I only change the order of definition

Reply:
I understand. Do you know why we do it this way, and whether it makes sense to update this?

Reply:
It seems we only require a merchant field if the participant is a policy expense chat. I think we need to confirm which cases we need to require merchant before updating

Reply:
So, do you know why we are passing participants in transaction, and why we have isPolicyExpenseChat in participant?

Reply:
I think I understand better now, and `transaction` isn't a Transaction like in the database, it just a holder of some data, and in this case it sounds like it can be `data related to Money Request view state, rather than the underlying Money Request data`

So what's the type of `transaction` here? 
- The props define it as `transactionPropTypes` which afaict doesn't have `participants`
- But I see ReportUtils.getTransactionDetails takes `OnyxEntry<Transaction>` and the `Transaction` type does have participants